src/application/usecases/calculate_exercise_statistics_use_case.ts

```diff
@@ -0,0 +1,171 @@
+import { queryStatsDict } from 'types/type_alias';
+import { Statistics } from '../../domain';
+import { ISubmission, ISubmissionApi } from '../../infrastructure';
+import { IUseCase } from './i_use_case';
+
+export interface IGetAStatisticsRequest {
+  exercise_id: string;
+}
+export interface IGetAStatisticsResponse {
+  statistics:
+    | {
+        id: string;
+        average_time: number;
+        passed_total: {
+          passed: number;
+          total: number;
+        };
+        query_result: queryStatsDict;
+      }
+    | undefined;
+}
+
+export class GetStatisticsUseCase
+  implements IUseCase<IGetAStatisticsRequest, IGetAStatisticsResponse>
+{
+  private readonly submissions_api: ISubmissionApi;
+
+  constructor(submissions_api: ISubmissionApi) {
+    this.submissions_api = submissions_api;
+  }
+
+  public async do(
+    request: IGetAStatisticsRequest,
+  ): Promise<IGetAStatisticsResponse> {
+    const submissionsResult = await this.submissions_api.getSubmissions(
+      request.exercise_id,
+    );
+
+    const statistics = await this.calculateStatistics(submissionsResult);
+    if (!statistics) {
+      return {
+        statistics: undefined,
+      };
+    }
+
+    return {
+      statistics: {
+        id: request.exercise_id,
+        average_time: statistics.average_time,
+        passed_total: {
+          passed: statistics.passed_total[0],
+          total: statistics.passed_total[1],
+        },
+        query_result: statistics.query_results,
+      },
+    };
+  }
+
+  public async calculateStatistics(
```

Comment:
You know better than this - det her skal trinvist forsines og splittes ud i funktioner

Reply:
+ flere kommentarer og docstrings